examples/jn516x/rpl/border-router/border-router.c

```diff
@@ -0,0 +1,97 @@
+/*
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ * 3. Neither the name of the Institute nor the names of its contributors
+ *    may be used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE INSTITUTE AND CONTRIBUTORS ``AS IS'' AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED.  IN NO EVENT SHALL THE INSTITUTE OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+ * OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+ * SUCH DAMAGE.
+ *
+ * This file is part of the Contiki operating system.
+ *
+ */
+
+#include "contiki.h"
+#include "net/rpl/rpl.h"
+#include "tools/rpl-tools.h"
+
+#define DEBUG DEBUG_PRINT
+#include "net/ip/uip-debug.h"
+
+static uip_ipaddr_t prefix;
+static uint8_t prefix_set;
+
+PROCESS(border_router_process, "Border router process");
+
+AUTOSTART_PROCESSES(&border_router_process);
+
+/*---------------------------------------------------------------------------*/
+void
+request_prefix(void)
+{
+  /* mess up uip_buf with a dirty request... */
+  uip_buf[0] = '?';
+  uip_buf[1] = 'P';
+/*  uip_buf[2] = '\n'; */
+  uip_len = 2;
+  slip_send();
+  uip_len = 0;
+}
+/*---------------------------------------------------------------------------*/
+void
+set_prefix_64(uip_ipaddr_t *prefix_64)
+{
+  memcpy(&prefix, prefix_64, 16);
+  prefix_set = 1;
+}
+/*---------------------------------------------------------------------------*/
+PROCESS_THREAD(border_router_process, ev, data)
+{
+  static struct etimer et;
+
+  PROCESS_BEGIN();
+
+/* While waiting for the prefix to be sent through the SLIP connection, the future
+ * border router can join an existing DAG as a parent or child, or acquire a default
+ * router that will later take precedence over the SLIP fallback interface.
+ * Prevent that by turning the radio off until we are initialized as a DAG root.
+ */
+  prefix_set = 0;
+
+  PROCESS_PAUSE();
+
+  PRINTF("RPL-Border router started\n");
```

Comment:
I haven't read the full PR yet, but why is there another border router other than the one at examples/ipv6?